Firestore/Example/FuzzTests/FuzzingTargets/FSTFuzzTestTargets.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FSTFuzzTestSerializer_h
+#define FSTFuzzTestSerializer_h
+
+#import <Foundation/Foundation.h>
+
+namespace firebase {
+namespace firestore {
+namespace fuzzing {
+
+// Returns the location of the Serializer dictionary file.
+inline NSString *GetSerializerDictionaryLocation(NSString *resources_location) {
+  return [resources_location stringByAppendingPathComponent:@"Serializer/serializer.dictionary"];
+}
+
+// Returns the location of the Serializer corpus. This corpus is a special
+// case because we generate its binary protos during the build process.
+inline NSString *GetSerializerCorpusLocation(NSString *resources_location) {
+  return @"FuzzTestsCorpus";
+}
+
+// Fuzz-test the deserialization process in Firestore. The Serializer reads raw
+// bytes and converts them to a model object.
+int FuzzTestDeserialization(const uint8_t *data, size_t size);
```

Comment:
I understand your intention here, but I'd avoid this sort of configuration when one header declares functions which are then defined in various source files. Style guide also [discourages](https://google.github.io/styleguide/cppguide.html#Header_Files) this: _"In general, every .cc file should have an associated .h file"_.

I suggest creating a `FSTFuzzTestSerializer.h` header and moving this declaration there. Then, either `FSTFuzzTestsPrincipal.mm` would include several such headers, or else this header can umbrella-include them (but the former is more straightforward and achieves the same result).

Reply:
OK, I agree. I will have a separate header file for every fuzzing target.